src/env.h

```diff
@@ -401,7 +401,13 @@ class AsyncHooks : public MemoryRetainer {
   void grow_async_ids_stack();
 
   v8::Global<v8::Array> js_execution_async_resources_;
-  std::vector<v8::Local<v8::Object>> native_execution_async_resources_;
+
+  // TODO(@jasnell): Note that this is technically illegal use of
+  // v8::Locals, which aren't supposed to live beyond the handle
+  // scope in which they were created. This should likely be a
```

Comment:
Just a note -- this isn't problematic because of lifetimes, like this comment impies. We keep the entries in this object as a stack that grows and shrinks with the C stack, so the entries here *will* actually be in the right handle scopes. But the `Local` values themselves aren't kept on the stack here, which I guess is why this is problematic.

Reply:
Updated the comment to incorporate this additional context.